### PR TITLE
Jk/develop/gamma decay

### LINKED ISSUE #904: Gamma decay loss weights across forecast steps
### Is your feature request related to a problem? Please describe.

For the stable roll-out experiments, implement the possibility to add weights according to forecast step. 

### Describe the solution you'd like

For 8 forecast steps, these are the normalized weights for different gamma factors:
Image

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### Organisation

ECMWF

### PR DESCRIPTION
**Pull request by Jubeku**
## Description

This PR enables to provide a weighting function to distinctly weight forecast steps in the loss calculations. 
The implemented function is a gamma function with a scale factor. The name of the weighting function and the scale factor are provided in the `default_config.yaml`.
In principle other functions could be implemented within the same structure. 

## Issue Number
Closes #904 

Is this PR a draft? Mark it as draft.

## Checklist before asking for review

-   [ ] I have performed a self-review of my code
-   [ ] My changes comply with basic sanity checks:
      - I have fixed formatting issues with `./scripts/actions.sh lint`
      - I have run unit tests with `./scripts/actions.sh unit-test`
      - I have documented my code and I have updated the docstrings.
      - I have added unit tests, if relevant
-   [ ] I have tried my changes with data and code:
      - I have run the integration tests with `./scripts/actions.sh integration-test`
      - (bigger changes) I have run a full training and I have written in the comment the run_id(s): `launch-slurm.py --time 60`
      - (bigger changes and experiments) I have shared a hegdedoc in the github issue with all the configurations and runs for this experiments
-   [ ] I have informed and aligned with people impacted by my change:
    - for config changes: the MatterMost channels and/or a design doc
    - for changes of dependencies: the MatterMost software development channel

### CODE REVIEW

File: config/default_config.yml [hunk 1]
```diff
@@ -76,6 +76,10 @@ loss_fcts_val:
     - "mse"
     - 1.0
 
+timestep_weight:
```
Comment (sophie-xhonneux) [inline config/default_config.yml]: is this backwards compatible?

File: src/weathergen/train/loss.py [hunk 2]
```diff
@@ -147,3 +147,9 @@ def mse_channel_location_weighted(
 def cosine_latitude(stream_data, forecast_offset, fstep, min_value=1e-3, max_value=1.0):
     latitudes_radian = stream_data.target_coords_raw[forecast_offset + fstep][:, 0] * np.pi / 180
     return (max_value - min_value) * np.cos(latitudes_radian) + min_value
+
+
+def gamma_decay(forecast_steps, gamma):
```
Comment (sophie-xhonneux) [inline src/weathergen/train/loss.py:152]: maybe we can put this in a util file?

**Comment by Jubeku** [comment]
Forecast fine-tuning experiments: As expected, gamma decay helps to improve scores in the early steps as they are weighted more during rollout training. 

image
image
